src/backend/utils/activity/pgstat_function.c

```diff
@@ -80,18 +78,6 @@ pgstat_init_function_usage(FunctionCallInfo fcinfo,
 	PgStat_BackendFunctionEntry *pending;
 	bool		created_entry;
 
-	if (pre_function_call_hook && IsTransactionState())
```

Comment:
We should keep this part instead of deleting them all, add a dialect check for Tsql would be better

Reply:
Moved this code to tsql extension and called using hook

Reply:
We should keep those code block here ( and also for the pre_function_call_hook definition ) , since PG code will go through this code block, and it's not babelfish specific code.